chain/client/src/client.rs

```diff
@@ -1373,8 +1373,20 @@ impl Client {
         self.chain.blocks_delay_tracker.mark_chunk_completed(&chunk_header, StaticClock::utc());
         self.block_production_info
             .record_chunk_collected(partial_chunk.height_created(), partial_chunk.shard_id());
+
         persist_chunk(partial_chunk, shard_chunk, self.chain.mut_chain_store())
             .expect("Could not persist chunk");
+
+        // We process chunk endorsements that were blocked by not having chunk complete.
+        self.chunk_endorsement_tracker.process_pending_endorsements(&chunk_header).unwrap_or_else(
+            |_| {
+                panic!(
```

Comment:
I don't think we should panic here. The client_actor implementation of `ChunkEndorsementMessage` just logs the error and continues.

```
        if let Err(err) = self.client.process_chunk_endorsement(msg.0) {
            tracing::error!(target: "client", ?err, "Error processing chunk endorsement");
        }
```

In a separate comment, I mentioned, we can do the tracing within `process_pending_endorsements` and change signature to not return an error.